### PR TITLE
Expose `PrtlSeqFailure`'s constructors and improve its errors

### PR DESCRIPTION
**Pull request by Jimbo4350**

### CODE REVIEW

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Prtcl.hs [hunk 1]
```diff
@@ -216,15 +229,15 @@ prtlSeqChecks ::
   WithOrigin (LastAppliedBlock crypto) ->
   BHeader crypto ->
   m ()
-prtlSeqChecks lab bh =
+prtlSeqChecks lab currentBh =
```
Comment (JaredCorduan) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Prtcl.hs:232]: More descriptive names are great!  In general, though, we've tried to keep the same names between the spec and the code. And sometimes when there are lots of long names in the spec it gets hard to read. It's all a  balancing act! I certainly don't want to slow you down, though!

**Comment by nc6** [comment]
What, why? How does `cardano-node` need these? The CHAIN STS is used only for testing, it isn't called from consensus (which implements its own checks to do these things) - there should be no need for these things in `cardano-node`.

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Prtcl.hs (hunk 1 above)
Comment (Jimbo4350) [reply shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Prtcl.hs:232]: That's fair enough, I didn't realize the variable names were chosen to match the spec, I'll revert this change.

**Comment by Jimbo4350** [comment]
bors merge

**Pull request merged**